src/deploymentstatus.h

```diff
@@ -6,28 +6,50 @@
 #define BITCOIN_DEPLOYMENTSTATUS_H
 
 #include <chain.h>
+#include <versionbits.h>
 
 #include <limits>
 
+/** Global cache for versionbits deployment status */
+extern VersionBitsCache versionbitscache;
+
 /** Determine if a deployment is active for the next block */
 inline bool DeploymentActiveAfter(const CBlockIndex* pindexPrev, const Consensus::Params& params, Consensus::BuriedDeployment dep)
 {
     assert(Consensus::ValidDeployment(dep));
     return (pindexPrev == nullptr ? 0 : pindexPrev->nHeight + 1) >= params.DeploymentHeight(dep);
 }
 
+inline bool DeploymentActiveAfter(const CBlockIndex* pindexPrev, const Consensus::Params& params, Consensus::DeploymentPos dep)
+{
+    assert(Consensus::ValidDeployment(dep));
+    return ThresholdState::ACTIVE == VersionBitsState(pindexPrev, params, dep, versionbitscache);
+}
+
 /** Determine if a deployment is active for this block */
 inline bool DeploymentActiveAt(const CBlockIndex& index, const Consensus::Params& params, Consensus::BuriedDeployment dep)
 {
     assert(Consensus::ValidDeployment(dep));
     return index.nHeight >= params.DeploymentHeight(dep);
 }
 
+inline bool DeploymentActiveAt(const CBlockIndex& index, const Consensus::Params& params, Consensus::DeploymentPos dep)
+{
+    assert(Consensus::ValidDeployment(dep));
+    return DeploymentActiveAfter(index.pprev, params, dep);
+}
+
 /** Determine if a deployment is enabled (can ever be active) */
 inline bool DeploymentEnabled(const Consensus::Params& params, Consensus::BuriedDeployment dep)
 {
     assert(Consensus::ValidDeployment(dep));
     return params.DeploymentHeight(dep) != std::numeric_limits<int>::max();
 }
 
+inline bool DeploymentEnabled(const Consensus::Params& params, Consensus::DeploymentPos dep)
+{
+    assert(Consensus::ValidDeployment(dep));
+    return params.vDeployments[dep].nTimeout != 0;
```

Comment:
e3a93b5095522f9315942600cbeb394a902bc5ce: This is currently unused, so there are no bugs. However, I am wondering if the implementation is the most meaningful. This is the first place where a magic value of `0` for the timeout is defined. Be aware that nTimeout is mostly an unsanitized [1] "don't care value", as the enabled status is read from nStartTime (NEVER_ACTIVE).

Finally, I am wondering what the use of this helper might be. Maybe delete the function until there is a use case?

```cpp
template <class> inline constexpr bool ALWAYS_FALSE{false};
template <class Dep> bool DeploymentEnabled(const Consensus::Params& params, Dep dep)
{
static_assert(ALWAYS_FALSE<Dep>, "DeploymentEnabled only meaningful for buried deployments");
```

[1] There was a favour to not sanitize this value across all ST implementations. https://github.com/bitcoin/bitcoin/pull/21392#discussion_r600435983

Reply:
> This is the first place where a magic value of `0` for the timeout is defined.

`!= 0` matches the behaviour used for `IsScriptWitnessEnabled` prior to 0328dcdcfcb56dc8918697716d7686be048ad0b3 

> Finally, I am wondering what the use of this helper might be. Maybe delete the function until there is a use case?

Having it defined makes it (relatively) easy to switch `DEPLOYMENT_SEGWIT` back to a signaled deployment (ie updating consensus/params.h and chainparams.cpp) and checking that the change does what it's supposed to (ie not require updating validation.cpp etc).

> the enabled status is read from nStartTime (NEVER_ACTIVE).

I think it would make sense to switch it to this -- should be possible to have a "fake" NEVER_ACTIVE that still passes DeploymentEnabled by setting the starttime to greater than `uint32_t::max()`. (Might do it if there's another rebase needed, otherwise will leave for a later PR)

Reply:
Fixed in https://github.com/bitcoin/bitcoin/pull/22385